SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -739,4 +719,37 @@ QueryConstraint InterestFactory::CreateLevelConstraints() const
 	return LevelConstraint;
 }
 
+void InterestFactory::AddObjectToConstraint(UObjectPropertyBase* Property, uint8* Data, QueryConstraint& OutConstraint) const
+{
+	UObject* ObjectOfInterest = Property->GetObjectPropertyValue(Data);
+
+	if (ObjectOfInterest == nullptr)
+	{
+		return;
+	}
+
+	FUnrealObjectRef UnrealObjectRef = PackageMap->GetUnrealObjectRefFromObject(ObjectOfInterest);
+
+	if (!UnrealObjectRef.IsValid())
+	{
+		return;
+	}
+
+	QueryConstraint EntityIdConstraint;
+	EntityIdConstraint.EntityIdConstraint = UnrealObjectRef.Entity;
+	OutConstraint.OrConstraint.Add(EntityIdConstraint);
+}
+
+void InterestFactory::SetResultType(Query& OutQuery, const ResultType& InResultType)
+{
+	if (GetDefault<USpatialGDKSettings>()->bEnableResultTypes)
+	{
+		OutQuery.ResultComponentId = InResultType;
```

Comment:
Why not use an add here?
If we're sure that we want to overwrite, then we could use MoveTemp()

Reply:
Ah no can't use MoveTemp as you use the same source array often.